src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1381,6 +1389,141 @@ void java_bytecode_parsert::rclass_attribute(classt &parsed_class)
+          //                           CONSTANT_Long
+          //                           CONSTANT_Float
+          //                           CONSTANT_Double
+          //                           CONSTANT_MethodHandle
+          //                           CONSTANT_MethodType
+
+          // We read the three arguments here to see whether they correspond to
+          // our hypotheses for this being a lambda function entry.
+
+          u2 argument_index1 = read_u2();
+          u2 argument_index2 = read_u2();
+          u2 argument_index3 = read_u2();
+
+          // The additional arguments for the altmetafactory call are skipped,
+          // as they are currently not used. We verify though that they are of
+          // CONSTANT_Integer type, cases where this does not hold will be
+          // analyzed further.
+          bool recognized = true;
+          for(size_t i = 3; i < num_bootstrap_arguments; i++)
+          {
+            u2 skipped_argument = read_u2();
+            recognized |= pool_entry(skipped_argument).tag == CONSTANT_Integer;
+          }
+          if(!recognized)
+          {
+            debug() << "format of BootstrapMethods entry not recognized" << eom;
+            return;
+          }
+
+          const pool_entryt &interface_type_argument =
+            pool_entry(argument_index1);
+          const pool_entryt &method_handle_argument =
+            pool_entry(argument_index2);
+          const pool_entryt &method_type_argument = pool_entry(argument_index3);
+
+          if(
+            !(interface_type_argument.tag == CONSTANT_MethodType &&
+              method_handle_argument.tag == CONSTANT_MethodHandle &&
+              method_type_argument.tag == CONSTANT_MethodType))
+            return;
+
+          debug() << "INFO: parse lambda handle" << eom;
+          optionalt<lambda_method_handlet> lambda_method_handle =
+            parse_method_handle(method_handle_argument);
+
+          if(
+            lambda_method_handle.has_value() &&
+            lambda_method_handle->handle_type !=
+              method_handle_typet::LAMBDA_METHOD_HANDLE)
+          {
+            error() << "ERROR: could not parse lambda function method handle"
+                    << eom;
+          }
+          else
+          {
+            lambda_method_handle->interface_type =
+              pool_entry(interface_type_argument.ref1).s;
+            lambda_method_handle->method_type =
+              pool_entry(method_type_argument.ref1).s;
+            debug() << "lambda function reference "
+                    << id2string(lambda_method_handle->lambda_method_name)
+                    << " in class \"" << parsed_class.name << "\""
+                    << "\n  interface type is "
+                    << id2string(pool_entry(interface_type_argument.ref1).s)
+                    << "\n  method type is "
+                    << id2string(pool_entry(method_type_argument.ref1).s)
```

Comment:
Can you refactor the generation of this debugging text into a `std::string pretty()` method of `lambda_method_handlet` class? I think it would be useful to be able to generate this later whilst debugging lambda functionality, instead of just at the point where the lambda function is parsed.